src/actions/logout.tsx

```diff
@@ -0,0 +1,12 @@
+import { useHistory } from 'react-router-dom';
+import { auth } from '../firebase';
+
+const history = useHistory();
+
+const logout = () =>
+  auth
+    .signOut()
+    .then(() => history.push('/'))
+    .catch(error => alert(error.message));
```

Comment:
I think in catch we shouldn't use alert, I've never seen this kind of method for error handling. I think the best way for now is console.error or console.log wirh error.message
Or we could build global error boundary (something like context wrapper but for catching the errors which could be thrown in catch)